### ISSUE TITLE
Face recognition problems

### ISSUE THREAD

**Issue by dadaloop82** [roflcoopter/viseron#96]
Good morning!
I use viseron on my raspberry 4 with 2Gb of Ram and I must say it works really very well! Thank you!

Regarding the face recognition however I have some problems: I can't understand what DLIB wants to recognize the face.
Let me explain better:

If I make selfie with my cell phone (at any distance) and then I load them in script for learn "my shapes", it always tells me that it doesn't find any face to process. I have tried in many ways, but nothing.
If I take a picture of myself from facebook, it processes it correctly, but it is very difficult for it to recognize me.

Looking in the logs I see that he often recognizes "unknown" faces - I would be curious to understand what he sees and why it doesn't work with my photos. Are there any rules to be respected in order for DLIB to recognize your face?

I saw that a request has already been made to save unknown faces, maybe this can help me.

In the past I used a similar project (frigate) by blakeblackshear and I must say that in comparison frigate is much faster to recognize objects, although I think it uses the same object learning technology. Is there a particular reason?

Thanks!

**Comment by coyote1** [comment]
I also use / test both programs. I can confirm that the frigate is a little faster in recognition. I would also be interested in whether there is a reason

**Comment by roflcoopter** [comment]
Thanks for trying out Viseron!

Are you taking the pictures using your an iPhone? I too had issues with images which had too high resolution. I havent looked into why this is the case tho sadly.

For the face recognition to work properly you have to be looking directly towards the camera, or atleast in a similar direction as the trained images. I intent to implement face alignment, which means that an algorithm will be used to reshape the face into being viewed from the front. I believe this will help a little bit with accuracy.

Regarding the speed of object detection, it depends on your configuration.
By default Viseron will only look for objects every 1 seconds, based on the setting ```interval``` under ```object_detection```.
If you also use motion detection, the object will be delayed a bit further, i will try to explain below.

Given this configuration:
```yaml
motion_detection:
  interval: 1
  frames: 3
  trigger_detector: true
object_detection:
  interval: 1
```

This means that Viseron will only look for objects when motion is detected.
Viseron will look for motion at a 1 second interval, and motion will only be triggered if it persist for 3 frames,
Therefore, after 3 seconds, the object detector will trigger, so you are looking at a 3ish seconds delay.

If you instead would use this configuration:
```yaml
motion_detection:
  interval: 1
  frames: 3
  trigger_detector: false
object_detection:
  interval: 1
```

Object detection would run every second, even if there is no motion.
If you wanna make it even faster you can set interval to something even lower.
Say you have a camera which runs at 6 FPS and you want to look for objects in every frame you can use this:
```yaml
motion_detection:
  interval: 1
  frames: 3
  trigger_detector: true
object_detection:
  interval: 0.16 # to calulate this you take 1/fps, so given 6 fps 1/6=~0.16
```

TLDR: the speed highly depends on your configuration! Higher speed comes with more CPU usage of course

**Comment by coyote1** [comment]
Thank you very much for the detailed explanation.  I think, that helps me to adjust Viseron

**Comment by roflcoopter** [comment]
Closing as i believe this is resolved.
Feel free to reopen if that is not the case
